jiratools/__init__.py

```diff
@@ -5,110 +5,17 @@
-from typing import Any, Optional, List
 
-import jira
-import requests
+from .utils import load_config, DEFAULT_LINK_TYPE
+from .helpers import list_from_config
 
-
-REQUIRED_KEYS = ("JIRA_URL", "USERNAME", "PASSWORD", "DEFAULT_ASSIGNEE", "TEST_PROJECT")
-DEFAULT_LINK_TYPE = "relates to"
-
-CONFIG_FILENAME = os.path.join(os.path.expanduser("~"), "jira.config")
-SAMPLE_CONFIG_FILENAME = os.path.join(
-    os.path.dirname(os.path.abspath(__file__)), "jira.config.example"
-)
-CONFIG = None
-
-
-class ConfigNotFoundException(Exception):
-    """Exception for config failure."""
-
-    pass
-
-
-def _exit(status: int = 0, message: Optional[str] = None) -> None:
-    """
-    Exit the program and optionally print a message to standard error.
-
-    Args:
-        status: Exit code to use for exit (optional)
-        message: Message to print to standard error (optional)
-    """
-    if message:
-        print(message, file=sys.stderr)
-    sys.exit(status)
-
-
-def error_if(check: Any, status: Optional[int] = None, message: str = "") -> None:
-    """
-    Exit the program if a provided check is true.
-
-    Exit the program if the check is true.
-    If a status is provided,
-    that code is used for the exit code;
-    otherwise the value from the check is used.
-    An optional message for standard error can also be provided.
-
-    Args:
-        check: Anything with truthiness that can determine
-            if the program should exit or not
-        status: Exit code to use for exit
-        message: Message to print to standard error if check is True
-    """
-    if check:
-        _exit(status=status or check, message=message.format(check))
-
-
-def _load_config() -> None:
-    """
-    Load CONFIG_FILENAME into CONFIG.
-
-    If not already loaded; call before touching CONFIG
-
-    """
-    global CONFIG
-    if CONFIG is not None:
-        return
-
-    config = ConfigParser()
-    message = 'Config file "{}" {{}}'.format(CONFIG_FILENAME)
-    error_if(not os.path.exists(CONFIG_FILENAME), message=message.format("not found"))
-    config.read(CONFIG_FILENAME)
-    section_name = "jira"
-    error_if(
-        section_name not in config,
-        message=message.format('missing "{}" section'.format(section_name)),
-    )
-    missing_keys = [key for key in REQUIRED_KEYS if key not in config[section_name]]
-    missing_message = message.format(
-        'section "{}" missing keys: {{}}'.format(section_name)
-    )
-    error_if(missing_keys, status=1, message=missing_message)
-    CONFIG = config[section_name]
-
-
-def get_client() -> jira.JIRA:
-    """
-    Return a configured JIRA client.
-
-    Configured per the user's home directory ``jira.config`` file.
-
-    """
-    _load_config()
-    if not CONFIG:
-        raise ConfigNotFoundException
-    client = jira.JIRA(
-        CONFIG["JIRA_URL"],
-        basic_auth=(CONFIG["USERNAME"], CONFIG.get("PASSWORD", raw=True)),
-    )
-    return client
+from .comment import cli_add_comment
```

Comment:
Why all the `cli_` prefixes? Can't we assume everything is for the CLI since this is a CLI tool?

Reply:
I know of at least one repo that used some of these functions for non-clie tools...but it's an enterprise repo that might not be supported for much longer due to things outside of the maintainer's control.